contracts/debtAllocators/GenericDebtAllocator.sol

```diff
@@ -0,0 +1,211 @@
+        // Default max base fee to uint256 max
+        maxAcceptableBaseFee = type(uint256).max;
+    }
+
+    /**
+     * @notice Check if a strategies debt should be updated.
+     * @dev This should be called by a keeper to decide if a strategies
+     * debt should be updated and if so by how much.
+     *
+     * This cannot be used to withdraw down to 0 debt.
+     *
+     * @param _strategy Address of the strategy to check.
+     * @return . bool repersenting if the debt should be updated.
+     * @return . calldata that includes what the debt should be updated to.
+     */
+    function shouldUpdateDebt(
+        address _strategy
+    ) external view returns (bool, bytes memory) {
+        // Check the base fee isn't to high.
+        if (block.basefee > maxAcceptableBaseFee) {
+            return (false, bytes("Base Fee"));
+        }
+
+        // Cache the vault variable.
+        IVault _vault = IVault(vault);
+        // Retrieve the strategy specifc parameters.
+        IVault.StrategyParams memory params = _vault.strategies(_strategy);
+        // Make sure its an active strategy.
+        require(params.activation != 0, "!active");
+
+        // Get the strategy specific debt config.
+        Config memory config = configs[_strategy];
+        // Make sure we have a target debt.
+        require(config.targetRatio != 0, "!target");
+
+        // Get the target debt for the strategy based on vault assets.
+        uint256 targetDebt = Math.min(
+            (_vault.totalAssets() * config.targetRatio) / MAX_BPS,
+            // Make sure it is not more than the max allowed.
+            params.max_debt
+        );
+
+        // If we need to add more.
+        if (targetDebt > params.current_debt) {
+            // We can't add more than the current idle.
+            uint256 toAdd = Math.min(
+                targetDebt - params.current_debt,
+                // This may underflow if idle is less than the min.
+                // Thats okay since it should return false anyway.
+                _vault.totalIdle() - _vault.minimum_total_idle()
+            );
+
+            // If the amount to add is over our threshold.
+            if (toAdd > config.minimumChange) {
+                // Return true and the calldata.
+                return (
+                    true,
+                    abi.encodeCall(
+                        _vault.update_debt,
+                        (_strategy, params.current_debt + toAdd)
+                    )
+                );
+            }
+            // If target debt is lower than the current.
+        } else if (targetDebt < params.current_debt) {
+            // Find out by how much.
+            uint256 toPull = Math.min(
+                params.current_debt - targetDebt,
+                // Account for the current liquidity constraints.
+                IVault(_strategy).maxWithdraw(address(_vault))
+            );
+
+            // Check if its over the threshold.
```

Comment:
"it's"

Reply:
https://github.com/yearn/vault-periphery/pull/15/commits/21b8ff41c6b5848a617675584c28b81077eb389e